api/twitter/twitterApi.py

```diff
@@ -2,168 +2,210 @@
-    for id in uid_list:
-        data = api.request('users/show',  {'user_id': id})
-        results['response'][ str(id)] = __format_user_info( data.json())
-    return results
-
-def get_tweet_info( tweet_id):
-    """Request little tweet information, return a dictionary"""
-    data = api.request('statuses/lookup', {'id': tweet_id}).json()
-    if len(data) == 0:
-        return {"response": None}
-    else:
-        return __format_tweet_info(data)
-
-def get_retweeters( tweet_id):
-    """Request the 100 last retweet ids, return them as a list"""
-    data = api.request('statuses/retweeters/ids', { 'id': str(tweet_id)})
-    response = {}
-    response['response'] = data.json()['ids']
-    retweeters = response['response']
-    """change user_ids from num to string"""
-    for index, num in enumerate(retweeters):
-        retweeters[index] = str(num)
-    return response
-
-def get_tweet_data( tweet_id):
-    """Request full tweet information, including retweet and user information"""
-    url = "statuses/retweets/:%s" % tweet_id
-    data = api.request(url, {'count': 100}).json()
-    results = {}
-    if len(data) == 0:
-        results['response'] = []
-    else:
-        results['response'] = __format_tweet_data(data)
-    return results
-
-def get_user_timeline( user_id):
-    """Get the latest tweets of a user.
-       Returns up to 200 retweets in 4 categories."""
-
-    tweets = __request_user_timeline( user_id)
-    return tweets
- 
+from api.twitter.tokenProvider import Token
+
+class TracemapTwitterApi:
+
+    def __request_twitter(self, route: str, params: dict, route_extension=""):
+        if not hasattr(self, route):
+            token_instance = Token(route)
+            setattr(self, route, token_instance)
+        api = getattr(self, route).api
+        while True:
+            try:
+                response = api.request("%s%s" % (route, route_extension), params)
+            except Exception as exc:
+                print("Error while requesting Twitter: %s" % exc)
+                time.sleep(10)
+                continue
+            parsed_response = response.json()
+            if parsed_response == []:
+                return parsed_response
+            error_response = self.__check_error(api, parsed_response)
+            if error_response:
+                if error_response == 'continue':
+                    continue
+                else:
+                    print('###Error from Twitter:%s\n###Route requested:%s\n###Params:%s' % (error_response, route, params))
+                    return {}
+            else:
+                return parsed_response
+
+    def __check_error(self, api, response):
+        error_response = ""
+        if 'error' in response:
+            error_response = self.__check_twitter_error_code(
+                response["error"])
+        elif 'errors' in response:
+            error_response = self.__check_twitter_error_code(
+                response["errors"][0]["code"])
```

Comment:
What about the case that there is more than one error in `response["errors"]`?

Reply:
This should also be part of #104

Reply:
Despite of using print or logging.err, is twitter response sometimes containing more than one error?